From d6fc4154600e6d42bba36aa65e703d5da874b7fd Mon Sep 17 00:00:00 2001
From: David Korth <gerbilsoft@gerbilsoft.com>
Date: Sun, 20 Apr 2025 04:21:02 -0400
Subject: [PATCH] [cmake] gcc.cmake: Move -Wno-error=stringop-overread here.

The same error showed up on armhf and amd64 this time.

Since it's happening in LTO and not the main compiler step, the pragmas
likely aren't working properly. Use -Wno-error=stringop-overread instead.

(cherry picked from commit ec441b4cec6df0b4e785a0b4861d6388aefe742e)
---
 cmake/platform/gcc.cmake              | 2 +-
 src/libromdata/Console/iQuePlayer.cpp | 8 --------
 2 files changed, 1 insertion(+), 9 deletions(-)

diff --git a/cmake/platform/gcc.cmake b/cmake/platform/gcc.cmake
index c18e63d5b..b3b715296 100644
--- a/cmake/platform/gcc.cmake
+++ b/cmake/platform/gcc.cmake
@@ -276,7 +276,7 @@ ENDIF(CFLAG_OPTIMIZE_FTREE_VECTORIZE)
 # Add "-Werror" *after* checking for everything else.
 SET(RP_C_FLAGS_COMMON   "${RP_C_FLAGS_COMMON} -Werror")
 SET(RP_CXX_FLAGS_COMMON "${RP_CXX_FLAGS_COMMON} -Werror")
-SET(CFLAGS_WNO_ERROR -Wno-error=unknown-pragmas -Wno-error=address -Wno-error=attributes -Wno-error=unused-parameter -Wno-error=unused-but-set-variable -Wno-error=ignored-qualifiers -Wno-error=missing-field-initializers -Wno-error=unused-variable -Wno-error=unused-function -Wno-error=type-limits -Wno-error=empty-body -Wno-error=address-of-packed-member -Wno-error=shift-negative-value -Wno-error=clobbered -Wno-error=overloaded-virtual -Wno-error=header-hygiene)
+SET(CFLAGS_WNO_ERROR -Wno-error=unknown-pragmas -Wno-error=address -Wno-error=attributes -Wno-error=unused-parameter -Wno-error=unused-but-set-variable -Wno-error=ignored-qualifiers -Wno-error=missing-field-initializers -Wno-error=unused-variable -Wno-error=unused-function -Wno-error=type-limits -Wno-error=empty-body -Wno-error=address-of-packed-member -Wno-error=shift-negative-value -Wno-error=clobbered -Wno-error=overloaded-virtual -Wno-error=header-hygiene -Wno-error=cast-align -Wno-error=stringop-overread)
 FOREACH(FLAG_TEST ${CFLAGS_WNO_ERROR})
 	# CMake doesn't like certain characters in variable names.
 	STRING(REGEX REPLACE "/|:|=" "_" FLAG_TEST_VARNAME "${FLAG_TEST}")
